query/src/storages/stage/stage_source.rs

```diff
@@ -168,115 +121,23 @@ impl StageSource {
         }
     }
 
-    async fn initialize(&mut self, file_name: String) -> Result<()> {
-        let ctx = self.ctx.clone();
-        let stage = &self.table_info.stage_info;
-        let file_format = stage.file_format_options.format.clone();
-
-        let op = Self::get_op(&self.ctx, &self.table_info.stage_info).await?;
-        let path = file_name;
-        let object = op.object(&path);
-
-        // TODO(xuanwo): we need to unify with MultipartFormat.
-        let compression_algo = match stage.file_format_options.compression {
-            StageFileCompression::Auto => CompressAlgorithm::from_path(&path),
-            StageFileCompression::Gzip => Some(CompressAlgorithm::Gzip),
-            StageFileCompression::Bz2 => Some(CompressAlgorithm::Bz2),
-            StageFileCompression::Brotli => Some(CompressAlgorithm::Brotli),
-            StageFileCompression::Zstd => Some(CompressAlgorithm::Zstd),
-            StageFileCompression::Deflate => Some(CompressAlgorithm::Zlib),
-            StageFileCompression::RawDeflate => Some(CompressAlgorithm::Deflate),
-            StageFileCompression::Xz => Some(CompressAlgorithm::Xz),
-            StageFileCompression::Lzo => {
-                return Err(ErrorCode::UnImplement("compress type lzo is unimplemented"));
-            }
-            StageFileCompression::Snappy => {
-                return Err(ErrorCode::UnImplement(
-                    "compress type snappy is unimplemented",
-                ));
+    fn get_input_format(
+        format: &StageFileFormatType,
+        schema: DataSchemaRef,
+        format_settings: FormatSettings,
+    ) -> Result<Arc<dyn InputFormat>> {
+        let name = match format {
+            StageFileFormatType::Csv => "csv",
+            StageFileFormatType::Json => "ndjson",
```

Comment:
`json` to `ndjson` is expected?

Reply:
not really.

come from here https://github.com/datafuselabs/databend/blob/main/query/src/storages/stage/stage_source.rs#L126

Reply:
Please add a comment here for the explanation.

Reply:
I think we should add  StageFileFormatType::NDJson